build/rofl/manifest.go

```diff
@@ -0,0 +1,312 @@
+		}
+		if err = m.Validate(); err != nil {
+			f.Close()
+			return nil, fmt.Errorf("invalid manifest '%s': %w", fn, err)
+		}
+		m.sourceFn, _ = filepath.Abs(f.Name()) // Record source filename.
+
+		f.Close()
+		return &m, nil
+	}
+	return nil, fmt.Errorf("no ROFL app manifest found (tried: %s)", strings.Join(ManifestFileNames, ", "))
+}
+
+// Validate validates the manifest for correctness.
+func (m *Manifest) Validate() error {
+	if len(m.Name) == 0 {
+		return fmt.Errorf("name cannot be empty")
+	}
+
+	if len(m.Version) == 0 {
+		return fmt.Errorf("version cannot be empty")
+	}
+	if _, err := version.FromString(m.Version); err != nil {
+		return fmt.Errorf("malformed version: %w", err)
+	}
+
+	switch m.TEE {
+	case TEETypeSGX, TEETypeTDX:
+	default:
+		return fmt.Errorf("unsupported TEE type: %s", m.TEE)
+	}
+
+	switch m.Kind {
+	case AppKindRaw:
+	case AppKindContainer:
+		if m.TEE != TEETypeTDX {
+			return fmt.Errorf("containers are only supported under TDX")
+		}
+	default:
+		return fmt.Errorf("unsupported app kind: %s", m.Kind)
+	}
+
+	if err := m.Resources.Validate(); err != nil {
+		return fmt.Errorf("bad resources config: %w", err)
+	}
+
+	for name, d := range m.Deployments {
+		if d == nil {
+			return fmt.Errorf("bad deployment: %s", name)
+		}
+		if err := d.Validate(); err != nil {
+			return fmt.Errorf("bad deployment '%s': %w", name, err)
+		}
+	}
+	if _, ok := m.Deployments[DefaultDeploymentName]; !ok {
+		return fmt.Errorf("must define at least the '%s' deployment", DefaultDeploymentName)
+	}
+
+	return nil
+}
+
+// SourceFileName returns the filename of the manifest file from which the manifest was loaded or
+// an empty string in case the filename is not available.
+func (m *Manifest) SourceFileName() string {
+	return m.sourceFn
+}
+
+// Save serializes the manifest and writes it to the file returned by `SourceFileName`, overwriting
+// any previous manifest.
+func (m *Manifest) Save() error {
+	data, err := yaml.Marshal(m)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(m.SourceFileName(), data, 0o644) //nolint: gosec
+}
+
+// DefaultDeploymentName is the name of the default deployment that must always be defined and is
+// used in case no deployment is passed.
+const DefaultDeploymentName = "default"
+
+// Deployment describes a single ROFL app deployment.
+type Deployment struct {
+	// AppID is the Bech32-encoded ROFL app ID.
+	AppID string `yaml:"app_id,omitempty" json:"app_id,omitempty"`
+	// Network is the identifier of the network to deploy to.
+	Network string `yaml:"network" json:"network"`
+	// ParaTime is the identifier of the paratime to deploy to.
+	ParaTime string `yaml:"paratime" json:"paratime"`
+	// Admin is the identifier of the admin account.
+	Admin string `yaml:"admin,omitempty" json:"admin,omitempty"`
+	// Debug is a flag denoting whether this is a debuggable deployment.
+	Debug bool `yaml:"debug,omitempty" json:"debug,omitempty"`
```

Comment:
If people will be copying this for the Localnet, the debug should be visible in rofl.yaml, so they don't forget to change it.

```suggestion
	Debug bool `yaml:"debug" json:"debug,omitempty"`
```